### PR TITLE
Fix #18739, Make testset results printing better

### LINKED ISSUE #18739: Test printing refinement
followup to #17165 / #18738, cc @kshyatt

right now the base tests are printing both old as-you-go info, in addition to a new table so it looks like

```
     * dsp                   maxrss 3501.18 MB
     * examples              maxrss 3501.18 MB
     * compile               maxrss 3501.18 MB

Test Summary: |     Pass  Broken     Total
  Overall     | 22171317      28  22171345
Test:                 | Total time (s): | GC time (s): | Percent in gc: | Allocated (MB): | RSS (MB):
linalg/triangular     |  546.77         |   11.92      |    2.18        | 8420.77         |  680.38
linalg/qr             |   89.66         |    1.85      |    2.06        | 2312.61         |  751.55
```

I propose we get rid of the old style and attempt to make the table just print new rows as they come in, if that's possible. The maxrss numbers are the same so it's duplicated information.

Additional printing-related followup from https://github.com/JuliaLang/julia/pull/17165#issuecomment-250615742, printing of `@test true` or `@test 1==1` should be adjusted to not bother showing `Expression: nothing` now that passing tests don't save the input expressions any more.

### PR DESCRIPTION
**Pull request by kshyatt**
Now results come in one by one and print their memory/timing data
as they come in. We will no longer print rss results twice. Also,
each test now prints which worker it ran on.

Here are some screencaps:
screen shot 2016-12-07 at 14 55 47
screen shot 2016-12-07 at 14 46 30

### CODE REVIEW

**Comment by yuyichao** [comment]
Does it print less info during test run and more info after all the tests are finished?

**Comment by kshyatt** [comment]
It prints the GC time etc info during the test run and less after all are finished.

**Comment by yuyichao** [comment]
(The Appveyor log is better since it has time stamp for each lines.) The new output lgtm. Just want to make sure the info are printed as early as possible and this is particularly important when there's crashes......

**Comment by StefanKarpinski** [comment]
Can we display test results as animated gifs in teh terminal?

**Comment by kshyatt** [comment]
@tkelman this ok with you as well?

**Comment by tkelman** [comment]
The other part of #18739 about "Expression: nothing" is not fixed here. And it would be good to shorten the titles of any columns we can so the minimum terminal width where this looks okay isn't so wide. But yeah tgis is most of it, thanks!

**Comment by KristofferC** [comment]
Looks very nice but +1 for shortening titles as much as possible.

**Comment by kshyatt** [comment]
I've fixed the two things you mentioned, @tkelman. Here are some photos:

screen shot 2016-12-08 at 10 53 56
screen shot 2016-12-08 at 10 54 25

**Comment by kshyatt** [comment]
The alignment looks a bit funky because I took some padding off of `Alloc`, since the tests may use a lot of MB (see `abstractarray` vs `int` up there) and I don't know in advance. The way I had it before the alignment would get messed up.

**Review by tkelman — APPROVED** [review]

**Comment by tkelman** [comment]
there's a whitespace error. and might be worth dropping the trailing colons in the title row

**Comment by kshyatt** [comment]
The test build passed for me locally so I'm rerunning to see what's going on. Log of failed Mac build available [here](https://gist.github.com/kshyatt/cccf56486c3acdd5041be90c3a399df6).

File: base/test.jl [hunk 1]
```diff
@@ -139,9 +141,9 @@ type Broken <: Result
 end
 function Base.show(io::IO, t::Broken)
     print_with_color(:yellow, io, "Test Broken\n")
-    if t.test_type == :skipped
-        println(io, "  Skipped: ", t.orig_expr)
-    else
+    if t.test_type == :skipped && !(t.orig_expr === nothing)
+        print(io, "  Skipped: ", t.orig_expr)
```
Comment (kshyatt) [inline base/test.jl]: oh god sorry will fix

**Comment by kshyatt** [comment]
This is weird. Why is `replutil` failing *here* and not on other PRs?

**Comment by kshyatt** [comment]
Thanks to @yuyichao I fixed the failing `replutil` test

**Comment by kshyatt** [comment]
Can I get another review @tkelman (or @yuyichao will you sign off on this?)

**Comment by KristofferC** [comment]
Maybe remove the colons line Tony suggested. Not a big deal but would make it even narrower.

**Comment by kshyatt** [comment]
- Colons gone
- CI passed
- Fixed a typo
- Checked alignment for short and long test names
- Error handling isn't *completely* stupid

OK to merge?

**Comment by tkelman** [comment]
should probably be squaahed, but into 2 commits - one for the replutil fix, another for everything else

**Pull request merged**